lib/errors.ts

```diff
@@ -0,0 +1,50 @@
+export class CallGraphGenerationError extends Error {
+  public innerError?: Error;
+
+  constructor(msg: string, innerError?: Error) {
+    super(msg);
+    Object.setPrototypeOf(this, CallGraphGenerationError.prototype);
+    this.innerError = innerError;
+  }
+}
+
+export class ClassPathGenerationError extends Error {
+  public innerError: Error;
+
+  constructor(innerError: Error) {
+    super('Class path generation error');
+    Object.setPrototypeOf(this, ClassPathGenerationError.prototype);
+    this.innerError = innerError;
+  }
+}
+
+export class EmptyClassPathError extends Error {
```

Comment:
From a logging standpoint, would we be able to aggregate the errors by their class? answering questions like how many `EmptyClassPathError` we saw in the last days?